types/block.go

```diff
@@ -56,6 +56,20 @@ func CalculateCoinbase(height BlockHeight) Currency {
 	return NewCurrency64(base).Mul(SiacoinPrecision)
 }
 
+// CalculateNumSiacoins calculates the number of siacoins in circulation at a
+// given height.
+func CalculateNumSiacoins(height BlockHeight) Currency {
+	deflationBlocks := BlockHeight(InitialCoinbase - MinimumCoinbase)
+	avgDeflationSiacoins := CalculateCoinbase(0).Add(CalculateCoinbase(height)).Div(NewCurrency64(2))
+	if height <= deflationBlocks {
+		deflationSiacoins := avgDeflationSiacoins.Mul(NewCurrency64(uint64(height + 1)))
+		return deflationSiacoins
+	}
+	deflationSiacoins := avgDeflationSiacoins.Mul(NewCurrency64(uint64(deflationBlocks + 1)))
+	trailingSiacoins := NewCurrency64(uint64(height - deflationBlocks)).Mul(CalculateCoinbase(height))
+	return deflationSiacoins.Add(trailingSiacoins)
```

Comment:
ah, didn't see that this was being called in the consistency check.
Still, as a matter of practice it's prudent to at least verify that the naive version is slow before taking the time to optimize it. My guess is that the run time of the naive version would be a small percentage of the total run time of `checkConsistency`.